### ISSUE TITLE
Remove deprecated Lambda runtimes

### ISSUE THREAD

**Issue by matsev** [aws-cloudformation/cfn-python-lint#898]
*cfn-lint version: (`cfn-lint --version`)*
cfn-lint 0.20.1

*Description of issue.*
Some of the Lambda runtimes that are marked as valid by cfn-lint are no longer supported by CloudFormation, e.g. `nodejs4.3`. Please see the [Lambda runtime](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-runtime) documentation for allowed values and the [Runtime Support Policy](https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html) page for deprecation schedule.

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)

**Comment by rhbecker** [comment]
Similarly, I get errors when declaring the newly available `nodejs10.x`, even after I execute `cfn-lint -u` ...

> W2030 You must specify a valid allowed value for LambdaRuntime (nodejs10.x).
> Valid values are ['dotnetcore1.0', 'dotnetcore2.0', 'dotnetcore2.1', 'go1.x', 'java8', 'nodejs', 'nodejs4.3-edge', 'nodejs4.3', 'nodejs6.10', 'nodejs8.10', 'provided', 'python2.7', 'python3.6', 'python3.7', 'ruby2.5']

**Comment by fatbasstard** [comment]
@rhbecker 10 is already added (https://github.com/aws-cloudformation/cfn-python-lint/issues/896 and https://github.com/aws-cloudformation/cfn-python-lint/pull/890) and will be available next release.

**Comment by rhbecker** [comment]
ah, didn't dig deep enough ... thanks, @fatbasstard

**Comment by fatbasstard** [comment]
@rhbecker No problem, we need to get a new release out ASAP...  Working on it!

**Comment by fatbasstard** [comment]
@matsev Took some time to respond because I wanted to check something before "just removing it".

**A bit TL;DR, but my opinion/findings:**

We can/should not remove these values from the "Allowed Values" rule. This because the [runtime lifecycle is a bit special](https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html):

_Let's use  `Node.js 6.10` as an example_

**Deprecated (Create): April 30, 2019** 
From this day _NEW_ Lambda functions cannot be created with this runime. Existing functions can still be updated and deployed. So this is easy: It can still be used in CloudFormation templates.

**Deprecated (Update): June 30, 2019**
From this day existing Lambda function cannot be updated anymore. And this is where caveat is. You cannot _update the function_ itself anymore. But: The rest of the stack can still be updated and deployed. 
If we would remove it from the allowed Values the Template would throw an `E3030` error. Which is "wrong" since the template could be perfectly deployment. As long as the function is not changed.

For people, teams and companies actively developing in CloudFormation removing the values seems like an obvious change, but because we/I are not the only people this seems invalid.

**Suggestion: New Rule**
But this is a valid use case, so what about a new rule? Create a `Warning` level ("this could possible break") rule to check for Lambda runtimes that are deprecated?
Keep a list of runtimes and the date in which they are deprecated (for update). If a template contains a runtime that passed the deprecation date: Throw an error that the runtime is deprecated.

Would be a first "time aware" rule, but I'm not seeing any direct issues.

Advantages: 
* Clear specific rule for runtime deprecation that is not "linked" to a new deployment of cfn-lint (due to the time awareness)
* No false-positives in the allowed-values rule (which is an Error level ("blocking, WILL break deployment") rule).
* It's actually functionally different then allowed Values, so seperate rule
* By making it a separate rule it can be disable by cfn-lint users that want to keep running deprecated functions

Is this workable for you @rhbecker ?

**Comment by matsev** [comment]
The case with `Deprecated (Create)` can also be wrong. For example, I may add a Lambda function to an existing CloudFormation template or I may create a new template based on some code with legacy runtime that I have found online. These templates will (presumably) fail during deployment since CloudFormation will refuse to create new Lambda instances.

IMHO, I would like cfn-lint to raise:
* an `error` if a Lambda runtime has been marked as `Deprecated (Update)`
* a `warning` if a Lambda runtime has been marked as `Deprecated (Create)`

Additionally, perhaps implement a flag to ignore such warnings and errors to handle the case you mentioned were a CloudFormation has been updated but contains an unchanged Lambda function that has been deployed earlier.

For a start, a minimalistic implementation would be to raise errors for runtimes that are deprecated both for update and create (e.g. essentially removing `nodejs`, `nodejs4.3` and `nodejs4.3-edge` from the list of [white listed Lambda runtimes](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/ExtendedSpecs/all/03_value_types.json#L1500-L1502)).
